core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -171,11 +165,14 @@ public void doWork() {
             log.error("Failed to add record from the Kafka topic" + topic + " the local store");
           }
         }
+
+        if (commitInterval > 0 && System.currentTimeMillis() - lastCommitTime > commitInterval) {
+          log.debug("Committing offsets");
+          consumer.commitSync();
+        }
       }
-    } catch (ConsumerTimeoutException cte) {
-      // Expect ConsumerTimeout == -1, so this *should* never happen
-      throw new IllegalStateException(
-          "KafkaStoreReaderThread's ConsumerIterator timed out despite expected infinite timeout.");
+    } catch (WakeupException we) {
+      this.consumer.close();
```

Comment:
It's probably because you also have a close() that follows the call to wakeup(). If you have close() in the finally block, then maybe you can get rid of the other call.